app/scripts/models/auth_brokers/fx-desktop.js

```diff
@@ -93,7 +93,10 @@ define([
       var channel = this._channel || new FxDesktopChannel();
 
       channel.init({
-        window: this.window
+        window: this.window,
+        // The browser is the origin. Events from the browser have no origin.
```

Comment:
Hmm, it sound like `isTrusted` isn't *quite* right then.

ISTM that the string value `"null"`, while fragile, will at least fail securely - if we happen to see it, then we're sure the event came from trusted desktop code, right?  So a "null OR expected-origin" check, although ugly, would be safe.